frontend/lib/src/AppNode.ts

```diff
@@ -499,30 +493,20 @@ export class BlockNode implements AppNode {
       // Otherwise, we are currently running a fragment, and our behavior
       // depends on the fragmentId of this BlockNode.
 
-      if (this.fragmentId) {
-        if (!fragmentIdsThisRun.includes(this.fragmentId)) {
-          // This BlockNode corresponds to a different fragment, so we know we
-          // won't be modifying it and can return early.
-          return this
```

Comment:
we are removing this short-cut here in order to detect stale widgets in children. Since this path is only used within fragments, I believe it shouldn't make a big performance difference. When a parent block was modified / run, the children are automatically invalidated by returning `undefined`, so there might be a difference when we have deep nestings of containers and the inner-ones are run, but I don't see another good generic way of how to find stale children in our current execution model.